File: Builder/PizzaExample/BuilderPizza/Director.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuilderPizza
{
    public class Director
    {
        private IPizzaBuilder builder;

        public IPizzaBuilder Builder
        {
            set { builder = value; }
        }

        public void BuildMargheritaPizza()
        {
            builder.Reset();
            builder.SetDough("Thin crust");
            builder.SetSauce("Tomato");
            builder.SetCheese("Mozzarella");
            builder.SetTopping("Basil");
            builder.SetSize(30);
        }

        public void BuildPepperoniPizza()
        {
            builder.Reset();
            builder.SetDough("Thick crust");
            builder.SetSauce("Tomato");
            builder.SetCheese("Cheddar");
            builder.SetTopping("Pepperoni");
            builder.SetSize(35);
        }
    }
}

File: Builder/Java/Pizza/src/Director.java
public class Director {
    private IPizzaBuilder builder;

    public void setBuilder(IPizzaBuilder builder) {
        this.builder = builder;
    }

    public void buildMargheritaPizza() {
        builder.reset();
        builder.setDough("Thin crust");
        builder.setSauce("Tomato");
        builder.setCheese("Mozzarella");
        builder.setTopping("Basil");
        builder.setSize(30);
    }

    public void buildPepperoniPizza() {
        builder.reset();
        builder.setDough("Thick crust");
        builder.setSauce("Tomato");
        builder.setCheese("Cheddar");
        builder.setTopping("Pepperoni");
        builder.setSize(35);
    }
}
